suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,250 @@
+
+const availableNetworkAccountTypes: Map<NetworkSymbol | undefined, AccountType[]> = new Map([
+    ['btc', ['normal', 'taproot', 'segwit', 'legacy']],
+    ['ltc', ['normal', 'segwit', 'legacy']],
+    ['btg', ['normal', 'segwit', 'legacy']],
+    ['dgb', ['normal', 'segwit', 'legacy']],
+    ['vtc', ['normal', 'segwit', 'legacy']],
+]);
+
+export const useAddCoinAccount = () => {
+    const dispatch = useDispatch();
+    const { translate } = useTranslate();
+    const openLink = useOpenLink();
+    const areTestnetsEnabled = useSelector(selectAreTestnetsEnabled);
+    const supportedNetworks = useSelector((state: DeviceRootState) =>
+        selectDiscoverySupportedNetworks(state, areTestnetsEnabled),
+    );
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccounts(state),
+    );
+    const device = useSelector(selectDevice);
+    const { showAlert, hideAlert } = useAlert();
+    const navigation =
+        useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AccountDetail>>();
+    const [availableCardanoTypes, setAvailableCardanoTypes] = useState<AccountType[]>([]);
+    const [supportedNetworkSymbols, setSupportedNetworkSymbols] = useState<NetworkSymbol[]>([]);
+    const [addedAccount, setAddedAccount] = useState<Account | undefined>(undefined);
+    const [isAddingCoinAccount, setIsAddingCoinAccount] = useState<boolean>(false);
+
+    useEffect(() => {
+        const getSupportedCardanoTypes = async ({
+            deviceState,
+            trezorDevice,
+        }: {
+            deviceState: string;
+            trezorDevice: TrezorDevice;
+        }) => {
+            const types = await dispatch(
+                getCardanoSupportedAccountTypesThunk({
+                    deviceState,
+                    device: trezorDevice,
+                }),
+            ).unwrap();
+
+            if (types.success && types.payload) {
+                setAvailableCardanoTypes(types.payload ?? []);
+            } else {
+                setAvailableCardanoTypes([]);
+            }
+        };
+
+        if (device?.state !== undefined) {
+            // Cardano supported account types can vary based on device and firmware, so we need to fetch them
+            getSupportedCardanoTypes({ deviceState: device.state, trezorDevice: device });
+        } else {
+            setAvailableCardanoTypes([]);
+        }
+    }, [device, dispatch]);
+
+    useEffect(() => {
+        const symbols: NetworkSymbol[] = [];
+        supportedNetworks.forEach(element => {
+            if (symbols.indexOf(element.symbol) === -1) {
+                symbols.push(element.symbol);
+            }
+        });
+
+        setSupportedNetworkSymbols(symbols);
+    }, [supportedNetworks]);
+
+    const getAvailableAccountTypesForNetwork = useCallback(
+        ({ network }: { network: Network }) =>
+            network.networkType === 'cardano'
+                ? availableCardanoTypes
```

Comment:
Directly get the available cardano types here and get rid of the useEffect?

Reply:
73bb8fb859